### DISCUSSION TITLE
Tally data out of TallyArbiter seems to be bottlenecked

### DISCUSSION THREAD

**Discussion by JTF4** [josephdadams/TallyArbiter#419] [General]
For awhile I've been trying to track down why tally data seems to be delayed or non responsive at times on our Ross Carbonite. 

After a bunch of debugging and console logs I think I've narrowed down the issue to when TallyArbiter sends tally data back out. When you console log all of the processing being done, it's instant, but when you look at where the tally data is being received it seems to hit a slow / non responsive period every few seconds. If your looking at a socketio client, it looks like the tally hangs, then jumps to the right value. On a TSL client, the tally hangs, then catches up before moving on to new data.

@hrueger @MatteoGheza do you have any idea what could be causing a bottle neck like this off the top of your head?

**Comment by hrueger** [comment]
Hm, the only thing I can think of here is nagle's algorithm, but I think it is disabled in NodeJS by default...

**Comment by josephdadams** [comment]
Isn’t nagles disabled by default? Did we enable it?

**Comment by JTF4** [comment]
I don't believe we're using it, but could a module we're using be importing it?

**Comment by JTF4** [comment]
I did some digging on this and it appears that by default everything works. The issue starts when you add socketio listener clients. Once you do that, then all of the TSL clients start hanging. 


Is it possible that processing the listener client tally updates synchronously is causing the problem?
